docs/pages/contributing/documentation/how-to-contribute.mdx

```diff
@@ -1,168 +1,389 @@
+
+1. Add files and commit changes to your local branch periodically with commands similar to the following:
+
+   ```code
+   $ git add <new-file-name>                      # Add a specific new file to be committed
+   $ git add -A                                   # Add all changed files to the list of files to be committed
+   $ git commit -am "Fix or feature description"  # Add and commit changes with a comment
+   $ git commit -m "Fix or feature description"   # Commit changes already added
+   ```
+
+1. Push your branch to `gravitational/teleport` with a command similar to the following:
+   
+   ```code
+   $ git push origin my-branch
+   ```
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), select your branch as the branch to 
+merge into main, then click **New pull request**.
+
+   The continuous integration (CI) pipeline automatically assigns reviewers to your pull request.  
+
+1. Wait for the minimum required approvals, then merge your pull request as soon as possible.
+
+### External contributors
+
+If you are an external contributor, you must commit your changes to your branch and push your branch to a 
+fork of the `gravitational/teleport` repository.
+
+To push changes as an external contributor:
+
+1. Verify that you have an SSH key pair and have stored your public key in GitHub.
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), click **Fork**, then select **Create a new fork**.
+
+1. Verify the owner and repository name.
+
+1. Click **Create fork**.
+   
+   Alternatively, you can create a fork from the command-line by running a command similar to the following:
+   
+   ```code
+   $ git remote add fork ssh://git@github.com/my-user/teleport
+   ```
+
+   In this command, `my-user` represents your GitHub user name and `teleport` is the name of your Teleport repository fork.
+
+1. Pull all of the changes from the remote repository into your local fork by running the following command:
+   
+   ```code
+   $ git fetch origin
+   ```
+
+1. Push changes from your local branch to the remote fork of the [Teleport repository](https://github.com/gravitational/teleport) 
+by running a command similar to the following:
+   
+   ```code
+   $ git push --set-upstream fork my-branch
+   ```
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), select your branch as the branch to merge into main, 
+then click **New pull request**.
+
+   The continuous integration (CI) pipeline automatically assigns reviewers to your pull request. 
+   To ensure your pull request conforms to certain conventions and passes automated checks, it
+   might require a companion—or _buddy_—pull request that includes your commits to pass CI/CD validation. 
+   The companion pull request continues to identify you as the author of the changes committed.
+   To ensure that your PR or its companion PR is merged, you should respond to reviewer feedback in a timely manner. 
```

Comment:
I would change the order of these points.

External contributors won't have permissions to respond to feedback on the buddy PR, so they should first respond to feedback on their original PR, and when it is in a good enough state to be merged we will open the buddy PR.